app/src/main/java/net/skyscanner/backpack/demo/meta/StoriesRepository.kt

```diff
@@ -0,0 +1,103 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.demo.meta
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import org.jetbrains.annotations.TestOnly
+
+interface StoriesRepository {
+
+  fun allComponents(): List<Component>
+
+  fun screenshotStories(): List<Story>
+
+  fun storiesOf(component: Component, compose: Boolean): List<Story>
+
+  fun isComposeSupportedFor(component: Component): Boolean
+
+  @TestOnly
+  fun testStories(): List<Story>
+
+  companion object {
+
+    fun getInstance(): StoriesRepository = StoriesRepositoryImpl
+  }
+}
+
+private object StoriesRepositoryImpl : StoriesRepository {
+
+  private val generatedStories = Story.all()
+  private val testStories = generatedStories.filter { it.component.name == "TestComponent" }
+  private val allStories = generatedStories - testStories
+
+  private val map = allStories
+    .groupBy { it.component }
+    .filter { (_, stories) -> stories.isNotEmpty() }
+    .toSortedMap(compareBy { it.name })
+
+  private val components = map.keys.toList()
+
+  override fun allComponents() = components
+
+  override fun screenshotStories() = allStories.filter { it.isScreenshot }
+
+  override fun storiesOf(component: Component, compose: Boolean): List<Story> =
+    map[component]
+      ?.filter { it.isCompose == compose }
+      ?.sortedBy { it.name }
+      ?: emptyList()
+
+  override fun isComposeSupportedFor(component: Component): Boolean =
+    map[component]?.any { it.isCompose } ?: false
+
+  override fun testStories() = testStories
+}
+
+@Component(name = "TestComponent", link = "test")
```

Comment:
what is the `link` used  for again?

Reply:
For instance, for lint rules in the future